src/content/content.service.ts

```diff
@@ -1,28 +1,119 @@
 import { Injectable } from '@nestjs/common'
-import { CreateContentDto } from './dto/create-content.dto'
-import { UpdateContentDto } from './dto/update-content.dto'
+import { PrismaService } from 'src/prisma/prisma.service'
+import { ContentfulService } from '../contentful/contentful.service'
+import { ContentType } from '@prisma/client'
+import { difference, intersection } from '../shared/util/sets.util'
+import { InputJsonObject } from '@prisma/client/runtime/library'
+
+// we have to do this for TypeScript enums 😢
+const CONTENT_TYPE_MAP: { [key: string]: ContentType } = {
+  aiChatPrompt: ContentType.aiChatPrompt,
+  aiContentTemplate: ContentType.aiContentTemplate,
+  articleCategory: ContentType.articleCategory,
+  blogArticle: ContentType.blogArticle,
+  blogHome: ContentType.blogHome,
+  blogSection: ContentType.blogSection,
+  candidateTestimonial: ContentType.candidateTestimonial,
+  election: ContentType.election,
+  faqArticle: ContentType.faqArticle,
+  faqOrder: ContentType.faqOrder,
+  glossaryItem: ContentType.glossaryItem,
+  goodPartyTeamMembers: ContentType.goodPartyTeamMembers,
+  onboardingPrompts: ContentType.onboardingPrompts,
+  pledge: ContentType.pledge,
+  privacyPage: ContentType.privacyPage,
+  promptInputFields: ContentType.promptInputFields,
+  redirects: ContentType.redirects,
+  teamMember: ContentType.teamMember,
+  teamMilestone: ContentType.teamMilestone,
+  termsOfService: ContentType.termsOfService,
+}
 
 @Injectable()
 export class ContentService {
-  create(createContentDto: CreateContentDto) {
-    console.log(`createContentDto =>`, createContentDto)
-    return 'This action adds a new content'
-  }
+  constructor(
+    private prisma: PrismaService,
+    private contentfulService: ContentfulService,
+  ) {}
 
   findAll() {
-    return `This action returns all content`
+    return this.prisma.content.findMany()
   }
 
   findOne(id: number) {
     return `This action returns a #${id} content`
   }
 
-  update(id: number, updateContentDto: UpdateContentDto) {
-    console.log(`updateContentDto =>`, updateContentDto)
-    return `This action updates a #${id} content`
+  private async getExistingContentIds() {
+    return new Set(
+      (
+        await this.prisma.content.findMany({
+          select: {
+            id: true,
+          },
+        })
+      ).map(({ id }) => id),
+    )
   }
 
-  remove(id: number) {
-    return `This action removes a #${id} content`
+  async syncContent(seed: boolean = false) {
+    const [entries = [], deletedEntries = []] =
+      await this.contentfulService.getSync(seed)
+    const recognizedEntries = entries.filter(
+      (entry) => CONTENT_TYPE_MAP[entry.sys.contentType.sys.id],
+    )
+    const entryIds = new Set(recognizedEntries.map((entry) => entry.sys.id))
+    const existingContentIds = await this.getExistingContentIds()
+    const existingEntries = intersection(existingContentIds, entryIds)
```

Comment:
why not use the build in function for sets?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/intersection